shared/p2p/service_test.go

```diff
@@ -70,15 +71,74 @@ func TestP2pPortTakenError(t *testing.T) {
 }
 
 func TestBroadcast(t *testing.T) {
-	s, err := NewServer(&ServerConfig{})
+	s, err := NewServer(&ServerConfig{Port: 12345})
+	if err != nil {
+		t.Fatalf("error while trying to create server: %s", err)
+	}
+
+	s2, err := NewServer(&ServerConfig{Port: 54321})
+	if err != nil {
+		t.Fatalf("error while trying to create server: %s", err)
+	}
+
+	s3, err := NewServer(&ServerConfig{Port: 11223})
 	if err != nil {
-		t.Fatalf("Could not start a new server: %v", err)
+		t.Fatalf("error while trying to create server: %s", err)
+	}
+
+	s.Start()
+	s2.Start()
+	s3.Start()
+
+	time.Sleep(time.Second * 5)
+
+	for _, peer := range s.host.Peerstore().Peers() {
+		err = s2.host.Connect(context.Background(), peerstore.PeerInfo{ID: peer, Addrs: s.host.Addrs()})
+		if err != nil {
+			t.Fatalf("error while trying to connect to peer: %s", err)
+		}
+
+		err = s3.host.Connect(context.Background(), peerstore.PeerInfo{ID: peer, Addrs: s.host.Addrs()})
+		if err != nil {
+			t.Fatalf("error while trying to connect to peer: %s", err)
+		}
+
+		break
 	}
 
 	msg := &shardpb.CollationBodyRequest{}
+
+	s.RegisterTopic("theTopic", msg)
+	s2.RegisterTopic("theTopic", msg)
+	s3.RegisterTopic("theTopic", msg)
+
+	s2Chan := make(chan Message)
+	s2.Subscribe(msg, s2Chan)
+
+	s3Chan := make(chan Message)
+	s3.Subscribe(msg, s3Chan)
+
+	time.Sleep(5 * time.Second)
```

Comment:
Why sleep?

Reply:
Those sleep is for some reason, if I dont sleep the subcribing was not ready, maybe is it because it launches a goroutine?